src/lib/utils.ts

```diff
@@ -260,21 +262,45 @@ export function encodeKey(
         return null;
       }
 
+      if (
+        typeof startingIndex !== 'number' ||
+        typeof totalArrayLength !== 'number'
+      ) {
+        throw new Error(
+          'Invalid startingIndex or totalArrayLength parameters. Values must be of type number.',
+        );
+      }
+
+      if (startingIndex < 0) {
+        throw new Error(
+          'Invalid startingIndex parameter. Value cannot be negative.',
```

Comment:
Suggest showing variable in quote in error string.

```suggestion
          'Invalid `startingIndex` parameter. Value cannot be negative.',
```

Reply:
I was not able to add brackets or double quotes in Error outputs. Tried both `"`, ```, `\`` due to the linter configuration.

@CJ42 do you have another idea to improve the outputs?

Reply:
Maybe if you escape them with backslash `\`. But it is a minor detail, it is fine as it is.

Reply:
As you see above, I tried backslash, but the linter removes it automatically. I don't know what's better- ignore formatting and linting for those lines or keep it as is. I would argue the second one is more convenient.

When I edit the error messages, the `throw` tests also start to fail:

<img width="911" alt="img1" src="https://github.com/ERC725Alliance/erc725.js/assets/61689369/20c0d327-6072-41ad-9816-8daa1bc6c296">

Reply:
You don't need a backslash for different quotes. What was the error when you put a plain backquote around the startingIndex and totalArrayLength? Usually error message don't contain back quotes, but I don't really see a reason why not.